consensus/src/dag/bootstrap.rs

```diff
@@ -527,10 +588,22 @@ impl DagBootstrapper {
                 new_round_tx,
                 self.epoch_state.clone(),
                 Duration::from_millis(round_state_config.adaptive_responsive_minimum_wait_time_ms),
-                leader_reputation_adapter.clone(),
             )),
         );
 
+        let chain_health: Arc<dyn TChainHealth> = match may_be_leader_reputation {
+            Some(adapter) => ChainHealthBackoff::new(
+                ChainHealthBackoffConfig::new(self.config.chain_backoff_config.clone()),
+                adapter.clone(),
+            ),
+            None => NoChainHealth::new(),
+        };
+        let pipeline_health = PipelineLatencyBasedBackpressure::new(
+            PipelineBackpressureConfig::new(self.config.pipeline_backpressure_config.clone()),
+            ordered_notifier.clone(),
+        );
+        let health_backoff =
+            HealthBackoff::new(self.epoch_state.clone(), chain_health, pipeline_health);
```

Comment:
none of these actually stops the progress fully right?

Reply:
correct, next PR stops voting.